### DISCUSSION TITLE
There is no registered service of type "BlazorBootstrap.IIdGenerator

### DISCUSSION THREAD

**Discussion by curt3105** [vikramlearning/blazorbootstrap#149] [Q&A]
Hi I am very new to blazorbootstrap and I have tripple checked the code and the installation but I keep getting this error; Unhandled exception rendering component: Cannot provide a value for property 'IdGenerator' on type 'BlazorBootstrap.Button'. There is no registered service of type 'BlazorBootstrap.IIdGenerator'.
System.InvalidOperationException: Cannot provide a value for property 'IdGenerator' on type 'BlazorBootstrap.Button'. There is no registered service of type 'BlazorBootstrap.IIdGenerator'.
   at Microsoft.AspNetCore.Components.ComponentFactory.

**Comment by gvreddy04** [comment]
The blazor bootstrap service registration is missing. 
Link: https://demos.blazorbootstrap.com/getting-started/blazor-server#register-services

![image](https://user-images.githubusercontent.com/2337067/217602948-70630422-7f40-4159-8178-b4f6ad715e4b.png)

**Getting started links:**
**WebAssembly** - https://demos.blazorbootstrap.com/getting-started/blazor-webassembly
**Server** - https://demos.blazorbootstrap.com/getting-started/blazor-server

**Reply by curt3105** [reply]
Perfect, I had that but it was in the server not in the client. Thank you!
